memlru/memlru.go

```diff
@@ -157,6 +158,21 @@ func (m *MemLRU) Delete(ctx context.Context, key string) error {
 	return nil
 }
 
+func (m *MemLRU) DeletePrefix(ctx context.Context, keyPrefix string) error {
+	m.mu.Lock()
+	for _, key := range m.backend.Keys() {
+		if _, ok := key.(string); !ok {
+			continue
+		}
+		if strings.HasPrefix(key.(string), keyPrefix) {
+			m.backend.Remove(key)
+		}
+	}
+	m.mu.Unlock()
```

Comment:
lets move this under m.mu.Lock() as line `defer m.mu.Unlock()` .. its just safer from human error